src/vcpkg/base/files.cpp

```diff
@@ -3812,6 +3824,39 @@ namespace vcpkg
 #endif // ^^^ !_WIN32
         }
 
+        int64_t file_time_now() const override
+        {
+#if defined(_WIN32)
+            return stdfs::file_time_type::clock::now().time_since_epoch().count();
+#else // ^^^ _WIN32 // !_WIN32 vvv
+            struct timespec ts;
+            clock_gettime(CLOCK_REALTIME, &ts);
+            return ts.tv_sec * 1'000'000'000 + ts.tv_nsec;
```

Comment:
```suggestion
            return int64_t{ts.tv_sec} * 1'000'000'000 + ts.tv_nsec;
```

Otherwise it's not necessarily safe to do that